src/main/java/org/mule/extension/db/internal/domain/connection/derby/DerbyConnectionProvider.java

```diff
@@ -91,63 +88,37 @@
   @ParameterGroup(name = CONNECTION)
   private DerbyConnectionParameters derbyParameters;
 
-  private DbConnectionProvider dbConnectionProvider;
-
   @Override
   public void initialise() throws InitialisationException {
-    dbConnectionProvider = new DbConnectionProvider(configName, registry, poolingProfile, columnTypes) {
-
-      @Override
-      public java.util.Optional<DataSource> getDataSource() {
-        return empty();
-      }
-
-      @Override
-      public java.util.Optional<DataSourceConfig> getDataSourceConfig() {
-        return ofNullable(derbyParameters);
-      }
-
-      @Override
-      protected DbConnection createDbConnection(Connection connection) throws Exception {
-        return new DerbyConnection(connection, dbConnectionProvider.resolveCustomTypes());
-      }
-
-      @Override
-      public java.util.Optional<DbError> getDbVendorErrorType(SQLException e) {
-        if (Arrays.stream(new String[] {FAILED_TO_START_DATABASE, NOT_FOUND}).anyMatch(e.getMessage()::contains)) {
-          return java.util.Optional.of(CANNOT_REACH);
-        }
-
-        return empty();
-      }
-
-    };
-
-    dbConnectionProvider.initialise();
+    super.columnTypes = columnTypes;
```

Comment:
this seems unnecesary. The SDK will pick up the `@Parameters` in the super class